app/Http/Controllers/Frontend/CategoryController.php

```diff
@@ -4,19 +4,31 @@
 
 use App\Repositories\Contracts\CategoryRepository;
 use App\Repositories\Contracts\ProductRepository;
+use App\Repositories\Contracts\PropertyRepository;
 
 class CategoryController extends FrontendController
 {
-    public function __construct(CategoryRepository $category)
+    protected $dataProperty = ['id', 'key', 'name'];
+    protected $propertyRepository;
+
+    public function __construct(CategoryRepository $category, PropertyRepository $property)
     {
         parent::__construct($category);
+        $this->propertyRepository = $property;
     }
 
     public function category($slug)
     {
         parent::category($slug);
         if ($this->compacts['item']->type == 'product') {
```

Comment:
if ($this->compacts['item']->type == 'product') {
=>
 if (isset($this->compacts['item']->type) && $this->compacts['item']->type == 'product') {

Để tránh xảy ra lỗi khi là $this->compacts['item'] empty